extensions/wikia/EmbeddableDiscussions/scripts/ext.embeddableDiscussions.js

```diff
@@ -10,14 +15,49 @@ require([
 		trackingMethod: 'analytics'
 	});
 
-	$(function() {
+	function getUiModalInstance() {
+		return uiFactory.init(['modal']);
+	}
+
+	function openModal(link, title) {
+		// Track impression
+		track({
+			action: tracker.ACTIONS.IMPRESSION,
+			label: 'embeddable-discussions-share-modal-loaded',
+		});
+
+		$.when(
+			getUiModalInstance()
+		).then(function (uiModal) {
```

Comment:
If you're not loading more than one request, this could just be: `uiFactory.init(['modal']).then(function (uiModal) {`